File: Data Structures & Algorithms/sum-of-two-integers/submission-0.cpp
class Solution {
public:
    int getSum(int a, int b) {
        bool x;
        bool y;
        bool z = 0;
        int r = 0;
        for(int i{0uz}; i < 32; i++){
            x = (a >> i) & 1;
            y = (b >> i) & 1;
            r += ((x ^ y) ^ z) << i;
            z = (x & y) | (z & (x ^ y));
        }

        return r;
    }
};

File: Data Structures & Algorithms/palindromic-substrings/submission-0.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        c = 0

        # odd
        for i in range(0, len(s)):
            j = i
            k = i
            while j >= 0 and k < len(s):
                if s[j] == s[k]:
                    c+=1
                else:
                    break
                j-=1
                k+=1

        # odd
        for i in range(0, len(s)):
            j = i
            k = i+1
            while j >= 0 and k < len(s):
                if s[j] == s[k]:
                    c+=1
                else:
                    break
                j-=1
                k+=1 

        return c